plasmapy/formulary/parameters.py

```diff
@@ -275,7 +281,8 @@ def ion_sound_speed(
     k: u.m ** -1 = None,
     gamma_e=1,
     gamma_i=3,
-    ion="p+",
+    *,
+    ion: Particle,
```

Comment:
This should look like....

```python
def ion_sound_speed(
    T_e: u.K,
    T_i: u.K,
    ion: Particle,
    n_e: u.m ** -3 = None,
    k: u.m ** -1 = None,
    gamma_e=1,
    gamma_i=3,
    z_mean=None,
) -> u.m / u.s:
```

`*`, overly simplified, indicates the end to the positional arguments, so keyword-only arguments should not be defined before it.

Reply:
So you are ok with changing the order of the arguments in this function call? I was using the `*` entry so that I could keep the order of arguments the same, but have `ion` (with no default) come after the other arguments which do have defaults.

Reply:
Keywords have no order associated with them, so doing this does not change the order at all.  It does make `ion` a required argument that is either positional or keyword.  So, the suggested makes the interface behave like...

```python
ion_sound_speed(2*u.eV, 3*u.eV, "He+")

# or

ion_sound_speed(2*u.eV, 3*u.eV, ion="He+")
```

All the other arguments are keyword-only, so they can be specified in any order.

Reply:
I'm not sure why you say that all other arguments are keyword-only. I just tried, and you can specify an electron density and k by just giving them as the next arguments. No keywords needed.
```python
>>> ion_sound_speed(2*u.eV, 3*u.eV, "He+", 1 * u.m**-3, 1 * u.m ** -1)
<Quantity 1.54899987 m / s>
```